mex.go

```diff
@@ -199,25 +315,30 @@ func (mexset *messageExchangeSet) newExchange(ctx context.Context, framePool Fra
 		msgID:     msgID,
 		ctx:       ctx,
 		recvCh:    make(chan *Frame, bufferSize),
+		errCh:     newErrNotifier(),
 		mexset:    mexset,
 		framePool: framePool,
 	}
 
 	mexset.Lock()
-	if existingMex := mexset.exchanges[mex.msgID]; existingMex != nil {
-		mexset.log.WithFields(
-			LogField{"msgID", mex.msgID},
-			LogField{"existingType", existingMex.msgType},
-			LogField{"newType", mex.msgType},
-		).Warn("Duplicate msg ID for active and new mex.")
+	addErr := mexset.addExchange(mex)
+	mexset.Unlock()
 
-		mexset.Unlock()
-		return nil, errDuplicateMex
-	}
+	if addErr != nil {
+		if addErr == errMexSetShutdown {
+			mexset.log.WithFields(
+				LogField{"msgID", mex.msgID},
+				LogField{"msgType", mex.msgType},
+			).Warn("Attempted to create new mex after mexset shutdown")
+		} else if addErr == errDuplicateMex {
+			mexset.log.WithFields(
+				LogField{"msgID", mex.msgID},
+				LogField{"newType", mex.msgType},
+			).Warn("Duplicate msg ID for active and new mex.")
+		}
```

Comment:
So the callers of this already log (specifically `inbound.go` and `connection.go`), these logs are here because we want to add more structured information that the caller won't know.